bin/alluxio-start.sh

```diff
@@ -179,112 +181,113 @@ run_safe() {
   done
 }
 
-while getopts "hNw" o; do
-  case "${o}" in
-    h)
-      echo -e "${USAGE}"
-      exit 0
-      ;;
-    N)
-      killonstart="no"
-      ;;
-    w)
-      wait="true"
+main() {
+  # get environment
+  get_env
+
+  # ensure log/data dirs
+  ensure_dirs
+
+  while getopts "hNw" o; do
+    case "${o}" in
+      h)
+        echo -e "${USAGE}"
+        exit 0
+        ;;
+      N)
+        killonstart="no"
+        ;;
+      w)
+        wait="true"
+        ;;
+      *)
+        echo -e "${USAGE}"
+        exit 1
+        ;;
+    esac
+  done
+
+  shift $((OPTIND-1))
```

Comment:
nit: shift $((${OPTIND} - 1)) is preferred.

Per google style guide:
# Prefer this:
addition=$((${X} + ${Y}))
substitution="${string/#foo/bar}"